app/src/main/java/com/example/calorietracker/FirstPageActivity.java

```diff
@@ -0,0 +1,42 @@
+package com.example.calorietracker;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.TextView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class FirstPageActivity extends AppCompatActivity {
+    private FirebaseAuth mAuth;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mAuth = FirebaseAuth.getInstance();
+        FirebaseUser currentUser = mAuth.getCurrentUser();
+        if (currentUser != null) {
+            reload(new DashboardActivity());
```

Comment:
Nu are sens sa faci acest apel. Mai bine procedezi invers: daca nu este autentificat dai finish la aceasta activitate si il muti pe user in LoginActivity